wallets/provider-all/src/index.ts

```diff
@@ -28,10 +32,26 @@ import * as trustwallet from '@rango-dev/provider-trustwallet';
 import * as walletconnect2 from '@rango-dev/provider-walletconnect-2';
 import * as xdefi from '@rango-dev/provider-xdefi';
 
-type Enviroments = Record<string, Record<string, string>>;
+import { isWalletConnectExcluded } from './helpers';
+
+interface Environments {
+  [key: string]: Record<string, string>;
+  walletconnect2: WalletConnectEnvironments;
+}
 
-export const allProviders = (enviroments?: Enviroments) => {
-  walletconnect2.init(enviroments?.walletconnect2 || {});
+export const allProviders = (
+  environments?: Environments,
+  selectedProviders?: (WalletType | ProviderInterface)[]
```

Comment:
I think you don't need to make a separate param. you can add it to `environments`. you can rename `environments` to `options`. I'm okay with both approach.